examples/custom_shaders_uniforms/Main.cc

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2015 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#if defined(__APPLE__)
+  #include <OpenGL/gl.h>
+  #include <GLUT/glut.h>
+#elif not defined(_WIN32)
+  #include <GL/glew.h>
+  #include <GL/gl.h>
+  #include <GL/glut.h>
+#endif
+
+#include <iostream>
+#include <vector>
+
+#include <ignition/common/Console.hh>
+#include <ignition/rendering.hh>
+
+#include "example_config.hh"
+
+#include "GlutWindow.hh"
+
+using namespace ignition;
+using namespace rendering;
+
+const std::string vertex_shader_file = "vertex_shader.glsl";
+const std::string fragment_shader_file = "fragment_shader.glsl";
+//! [init shaders variables]
+
+const std::string RESOURCE_PATH =
+    ignition::common::joinPaths(std::string(PROJECT_BINARY_PATH), "media");
+
+//////////////////////////////////////////////////
+void buildScene(ScenePtr _scene)
+{
+  // initialize _scene
+  _scene->SetAmbientLight(0.3, 0.3, 0.3);
+  VisualPtr root = _scene->RootVisual();
+
+  // create directional light
+  DirectionalLightPtr light0 = _scene->CreateDirectionalLight();
+  light0->SetDirection(-0.5, 0.5, -1);
+  light0->SetDiffuseColor(0.5, 0.5, 0.5);
+  light0->SetSpecularColor(0.5, 0.5, 0.5);
+  root->AddChild(light0);
+
+  // create shader materials
+  // path to look for vertex and fragment shader parameters
+  std::string vertex_shader_path = ignition::common::joinPaths(
```

Comment:
```suggestion
  std::string vertexShaderPath = ignition::common::joinPaths(
```

camel case for the variable names